dice.py

```diff
@@ -1,23 +1,88 @@
+
+Current limitations:
+* Each die needs to have the same number of sides
+* The graphics only display if ALL dice have a value less than or equal to 6
+* The number of dice and how many faces they have is currently hardcoded under configurables
+* The return_dice_as_text function returns "a 11" instead of "an 11". I really CBA to fix that
+"""
+
 import random
 
-repeat = True
-while repeat:
-    no = random.randint(1, 6)
-    print("You rolled", no)
-    if no == 1:
-        print("You feel somewhat disappointed")
-    if no == 2:
-        print("You look at the two dots and begin to mourn a lover lost at sea")
-    if no == 3:
-        print("Three is the magic number, you turn into a frog")
-    if no == 4:
-        print("4! You win...not really, you're just rolling a dice. Get some mates!")
-    if no == 5:
-        print("You bleed a little")
-    if no == 6:
-        print("You begin to wonder what this point of this is")
-    if no == 7:
-        print("err...what? This is literally impossible")
-
-    print("Do you want to roll again Y/N (please roll again, it...it pleases me?)")
-    repeat = "Y" in input()
\ No newline at end of file
+"""Configurables"""
+number_of_dice = 3
+faces = 6
+
+
+def main():
+    repeat = True
+    while repeat:
+        display(roll(number_of_dice, faces))
+        print("Do you want to roll again Y/N (please roll again, it...it pleases me?)")
+        repeat = "Y" in input().upper()
+
+
+def roll(number_of_dice, faces):
+    return [random.randint(1, faces) for n in range(number_of_dice)]
+
+
+def display(dice):
+    if all(number <= 6 for number in dice):
+        for number in dice:
+            if number == 1:
+                print(" -------")
+                print("｜     ｜")
+                print("｜  o  ｜")
+                print("｜     ｜")
+                print(" -------")
+            elif number == 2:
+                print(" -------")
+                print("｜o    ｜")
+                print("｜     ｜")
+                print("｜    o｜")
+                print(" -------")
+            elif number == 3:
+                print(" -------")
+                print("｜o    ｜")
+                print("｜  o  ｜")
+                print("｜    o｜")
+                print(" -------")
+            elif number == 4:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜     ｜")
+                print("｜o   o｜")
+                print(" -------")
+            elif number == 5:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜  o  ｜")
+                print("｜o   o｜")
+                print(" -------")
+            elif number == 6:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜o   o｜")
+                print("｜o   o｜")
+                print(" -------")
+    print("You rolled " + return_dice_as_text(dice))
+
+
+def return_dice_as_text(dice):
```

Comment:
So is this bit below supposed to show the total of the 3 dice rolls?

Reply:
This bit is probably me being too clever for my own good but I wanted to show you some f strings at work. When you run the code it tells you what each dice roll was. This function helps to turn that into a proper sentence. It adds commas between die scores and puts and 'and a' before the last one, so a possible output would be:
"You rolled a 4, a 3 and a 2"
If you just printed the list it would say
"You rolled 3, 4, 2"